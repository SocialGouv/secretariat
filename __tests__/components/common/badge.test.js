import { render } from "@testing-library/react"

import Badge from "../../../src/components/common/badge"

it("renders expiry badge", () => {
  const { container } = render(
    <Badge type="expiry" label="expiry" title="expiry" />
  )
  expect(container).toMatchSnapshot()
})

it("renders warning badge", () => {
  const { container } = render(
    <Badge type="warning" label="warning" title="warning" />
  )
  expect(container).toMatchSnapshot()
})

it("renders disabled badge", () => {
  const { container } = render(
    <Badge type="disabled" label="disabled" title="disabled" />
  )
  expect(container).toMatchSnapshot()
})

it("renders icon warning badge", () => {
  const { container } = render(<Badge type="warning" title="icon badge" />)
  expect(container).toMatchSnapshot()
})
