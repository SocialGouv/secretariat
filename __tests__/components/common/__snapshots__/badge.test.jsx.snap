// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders disabled badge 1`] = `
<div>
  <p
    class="badge disabled"
    title="disabled"
  >
    disabled
  </p>
</div>
`;

exports[`renders expiry badge 1`] = `
<div>
  <p
    class="badge expiry"
    title="expiry"
  >
    expiry
  </p>
</div>
`;

exports[`renders icon warning badge 1`] = `
<div>
  <p
    class="badge warning icon"
    title="icon badge"
  />
</div>
`;

exports[`renders warning badge 1`] = `
<div>
  <p
    class="badge warning"
    title="warning"
  >
    warning
  </p>
</div>
`;
